### DISCUSSION TITLE
How to add attributes (Discord ID) to session when using a database?

### DISCUSSION THREAD

**Discussion by Brawl345** [nextauthjs/next-auth#2496] [Help]
I'm really struggling on how to add the discord ID to the session so that I can use it client-side in React. I only get email, image and the username. I tired extending `session` via `callbacks` but the `user` I get there is just the user from the database and doesn't have the information I need. I think I also need the extend the user model, but I don't know how I get the info I want into it :/

Can someone help me please?

**Comment by StefanSelfTaught** [comment]
Check out the jwt and session callbacks and their available parameters, you can also access `profile` and `account` in the jwt callback.

https://next-auth.js.org/configuration/callbacks

**Reply by Brawl345** [reply]
Already tried that but it doesn't seem to work when using a database. It just doesn't get called. I worked around it now by sending the auto incrementing ID from the users table back in the session and work with it but it feels like it's not a good solution :/

**Reply by sastraxi** [reply]
I worked around the same issue in my project by simply using the `email` as a unique identifier for my secondary tables with non-auth-specific user information. It's not the cleanest; I'd rather use the generated PKs for speed (via Prisma adapter), but there does not appear to be a way to access these adapter-specific IDs via the exposed API.
